crdt.go

```diff
@@ -608,7 +611,10 @@ func (store *Datastore) processNode(ng *crdtNodeGetter, root cid.Cid, rootPrio u
 		if known {
 			// we reached a non-head node in the known tree.
 			// This means our root block is a new head.
-			store.heads.Add(root, rootPrio)
+			err = store.heads.Add(root, rootPrio)
+			if err != nil {
+				return nil, errors.Wrapf(err, "error adding head %s", root)
+			}
```

Comment:
Isn't this changing behaviour? Instead of continuing it will abort.

Reply:
Yes, that was deliberate.  If you tell me that this error should be ignored, I can change this to make that explicit.  Though it does seem more consistent with how we handle a failure in the call to `Replace`.

Reply:
Yes, I think better to continue. Good to warn about it.